src/bit-fields-macros/src/builder.rs

```diff
@@ -0,0 +1,830 @@
+                        }).map_err(bit_fields::TryFromFlagSetAndFieldMapError::FlagSet)?;
+
+                        let value = map.into_iter().try_fold(value,|mut acc,(k,v)|match k.to_string().as_str() {
+                            #field_matching_from_hashmap
+                            _ => Err(bit_fields::TryFromFieldMapError::UnknownField)
+                        }).map_err(bit_fields::TryFromFlagSetAndFieldMapError::FieldMap)?;
+
+                        #ending
+                    }
+                }
+            }
+        };
+
+        let binary_ops = if data_type.is_nonzero() {
+            quote! {
+                impl #struct_name {
+                    //
+                    unsafe fn unchecked_bitand(self, rhs: Self) -> Self {
+                        Self(#data_type::new_unchecked(self.0.get() & rhs.0.get()))
+                    }
+                    unsafe fn unchecked_not(self) -> Self {
+                        Self(#data_type::new_unchecked(!self.0.get()))
+                    }
+                }
+            }
+        } else {
+            quote! {
+                impl std::ops::BitAnd for #struct_name {
+                    type Output = Self;
+                    fn bitand(self, rhs: Self) -> Self::Output {
+                        Self(self.0 & rhs.0)
+                    }
+                }
+                impl std::ops::Not for #struct_name {
+                    type Output = Self;
+                    fn not(self) -> Self::Output {
+                        Self(!self.0)
+                    }
+                }
+            }
+        };
+
+        let equal_cmp = if data_type.is_nonzero() {
+            quote! { (self.0.get() & mask.0.get()) == (other.0.get() & mask.0.get()) }
+        } else {
+            quote! { (self.0 & mask.0) == (other.0 & mask.0) }
+        };
+
+        let display_impl = {
+            let display_full_string = display_string.to_string();
+            quote! {
+                impl std::fmt::Display for #struct_name {
+                    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+                        write!(f,#display_full_string,#display_fmt_string)
+                    }
+                }
+            }
+        };
+
+        let header = format!(
+            "A {} bit structure containing a number of bit flags and bit fields.",
+            self.data_type.size()
+        );
+
+        let empty = if data_type.is_nonzero() {
+            TokenStream::new()
+        } else {
+            quote! {
+                /// Alias for `Self(0)`.
+                pub const fn empty() -> Self {
+                    Self(0)
+                }
+            }
+        };
+
+        let (flag_mask, range_mask) = if data_type.is_nonzero() {
+            (
+                quote! { unsafe { #data_type::new_unchecked(#bit_flag_mask) } },
+                quote! { unsafe { #data_type::new_unchecked(#bit_range_mask) } },
```

Comment:
If the bit field does not contain bits or ranges respectively, these will compile to `NonZero*::new_unchecked(0)`, which is undefined behavior

Reply:
Fixed.